### DISCUSSION TITLE
Unsupported devices - update plannned

### DISCUSSION THREAD

**Discussion by MatthiasDVB** [make-all/tuya-local#2339] [Ideas]
Hi together, are there any plans to support newer switches like 3way wallmount switch with 3,5" touchscreen ?
![461420759_10224969254393701_5900920820421198467_n](https://github.com/user-attachments/assets/a23573fa-5c36-4fdb-82a4-3dc99d7bab6e)
This devices are beautiful and working perfect, but unsupported.

**Comment by make-all** [comment]
What does "the devices are working perfectly, but unsupported" mean? What is your expectation here?

**Comment by MatthiasDVB** [comment]
This device has good quality, looks nice works reliable from Smartlife/Tuya App and of course from Tuya IoT.
My expectation is to use this from Home Assistant, this device has 3 way switch and I hope in the future I can use the switches with HA.
If no one knows this item, it is a wall mount 3way switch with extra display for scenes, camera and more.
But using 3way switch with HA is perfect.

**Reply by make-all** [reply]
Unfortunately it takes more than hope to get devices supported.

This is a community driven project, and the developers have no special access to technical information from tuya or device manufacturers. We rely on owners of the devices to collect what technical information they can from the Home Assistant logs and iot.tuya.com and to provide the information in a New Device request (Issue), or read the documentation and write a config file themselves and submit as a pull request.

**Comment by MatthiasDVB** [comment]
OK thx for your reply, I'm just an End-User, so it was my fault thinking that this devices work with HA :)
I thought that a switch will work when Tuya protocol is used.

**Comment by make-all** [comment]
The Tuya local protocol is not well defined enough for devices to just work. There is some info available through the cloud API that can help with automatically identifying the purposes of the datapoints a device exposes, which is what the cloud tuya integration built into HA uses, and also I think localtuya cloud assisted setup uses this info too, but this integration was designed to not require cloud APIs for anything, so configs need to be manually defined. Recent changes to use cloud assisted setup are limited to getting local keys and identifying the devices attached to your account.

**Comment by MatthiasDVB** [comment]
Well, thank you for your comment and explanation.
I have most data collected, I have IP address, ID and Key, and I have the thre necessary Data Points recorded, but still doesn't work.
How can I manually add DPIs to Tuya Local ?
